cloud-core/src/main/java/cloud/commandframework/CommandTree.java

```diff
@@ -638,8 +637,7 @@ private CommandTree(final @NonNull CommandManager<C> commandManager) {
             }
             return Collections.emptyList();
         } else if (commandQueue.peek().isEmpty()) {
-            commandContext.setCurrentArgument(child.getValue());
-            return child.getValue().getSuggestionsProvider().apply(commandContext, commandQueue.remove());
+            return this.directSuggestions(commandContext, child, commandQueue.peek());
```

Comment:
the change is intentional, the removal was inconsistent across the different branches of the if-statement, and it resulted on differing behavior for no good reason. The remaining argument shouldn't be consumed off of the queue because the suggestion parser may want to access it, and although that's more related to #190 it's still a step towards enabling that to work more reliably.